src/lpython/semantics/python_comptime_eval.h

```diff
@@ -211,13 +211,19 @@ struct PythonIntrinsicProcedures {
         if (ASRUtils::is_integer(*type)) {
             int64_t a = ASR::down_cast<ASR::IntegerConstant_t>(arg1)->m_n;
             int64_t b = ASR::down_cast<ASR::IntegerConstant_t>(arg2)->m_n;
+            if(b == 0){
+                throw SemanticError("Integer division or modulo by zero",loc);
+            }
             return ASR::down_cast<ASR::expr_t>(
-                ASR::make_IntegerConstant_t(al, loc, a%b, type));
+                ASR::make_IntegerConstant_t(al, loc, ((a%b)+b)%b, type));
         } else if (ASRUtils::is_real(*type)) {
             double a = ASR::down_cast<ASR::RealConstant_t>(arg1)->m_r;
             double b = ASR::down_cast<ASR::RealConstant_t>(arg2)->m_r;
+            if (b == double(0)) {
+                throw SemanticError("Float division or modulo by zero", loc);
+            }
             return ASR::down_cast<ASR::expr_t>(
-                ASR::make_RealConstant_t(al, loc, std::fmod(a, b), type));
+                ASR::make_RealConstant_t(al, loc, std::fmod(std::fmod(a, b) + b, b), type));
```

Comment:
I guess same formula but with `std::fmod` calls. Will recommend the same here as my previous review for integers.

Reply:
N.B. - https://github.com/lcompilers/lpython/pull/1566/files#r1130429810

Reply:
I guess adding the link as comments once suffices. Do I need to add it above this snippet as well ?

Reply:
Add the link here as well.